### PR TITLE
tke-business supports registry-namespaces.

### PR DESCRIPTION
**Pull request by defydahui**
Mainly add a feature to support registry-namesapces.

### CODE REVIEW

File: api/business/v1/conversion.go [hunk 1]
```diff
@@ -91,3 +92,20 @@ func AddFieldLabelConversionsForPlatform(scheme *runtime.Scheme) error {
...
+
+// AddFieldLabelConversionsForImageNamespace adds a conversion function to convert
+// field selectors of ImageNamespace from the given version to internal version
+// representation.
+func AddFieldLabelConversionsForImageNamespace(scheme *runtime.Scheme) error {
+	return scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.WithKind("Namespace"),
+		func(label, value string) (string, string, error) {
+			switch label {
+			case "spec.tenantID",
+				"status.phase",
```
Comment (choujimmy) [inline api/business/v1/conversion.go]: missed spect.name to indexable?

File: api/business/types.go [hunk 2]
```diff
@@ -294,3 +296,73 @@ type ConfigMapList struct {
...
+}
+
+// ImageNamespaceSpec represents an image namespace.
+type ImageNamespaceSpec struct {
+	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage.
+	// +optional
+	Finalizers  []FinalizerName
+	Name        string
+	TenantID    string
+	DisplayName string
```
Comment (choujimmy) [inline api/business/types.go]: missed `// +optional` comment

File: api/business/v1/types.go [hunk 3]
```diff
@@ -299,3 +301,73 @@ type ConfigMapList struct {
...
+}
+
+// ImageNamespaceSpec represents an image namespace.
+type ImageNamespaceSpec struct {
+	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage.
+	// +optional
+	Finalizers  []FinalizerName `json:"finalizers,omitempty" protobuf:"bytes,1,rep,name=finalizers,casttype=FinalizerName"`
+	Name        string          `json:"name" protobuf:"bytes,2,opt,name=name"`
+	TenantID    string          `json:"tenantID" protobuf:"bytes,3,opt,name=tenantID"`
+	DisplayName string          `json:"displayName" protobuf:"bytes,4,opt,name=displayName"`
```
Comment (choujimmy) [inline api/business/v1/types.go]: missed `// +optional` comment and `,omitempty` field tag

File: pkg/business/controller/imagenamespace/imagenamespace_health.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	namespaceList, err := c.registryClient.Namespaces().List(metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.tenantID=%s,spec.name=%s", imageNamespace.Spec.TenantID, imageNamespace.Spec.Name),
+	})
+	if err != nil {
+		return err
+	}
+	if len(namespaceList.Items) == 0 {
+		return fmt.Errorf("imagenamespace %s in tenant %s not exist", imageNamespace.Spec.Name, imageNamespace.Spec.TenantID)
+	}
+	return nil
```
Comment (choujimmy) [inline pkg/business/controller/imagenamespace/imagenamespace_health.go]: Here you need to check whether the image namespace is locked in the registry api

File: pkg/business/registry/imagenamespace/strategy.go [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+// MatchNamespace returns a generic matcher for a given label and field selector.
+func MatchNamespace(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
+	return storage.SelectionPredicate{
+		Label:    label,
+		Field:    field,
+		GetAttrs: GetAttrs,
+		IndexFields: []string{
+			"spec.tenantID",
+			"metadata.name",
+		},
```
Comment (choujimmy) [inline pkg/business/registry/imagenamespace/strategy.go]: need add "spec.name"

File: pkg/business/registry/imagenamespace/strategy.go [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+	}
+}
+
+// ToSelectableFields returns a field set that represents the object
+func ToSelectableFields(imageNamespace *business.ImageNamespace) fields.Set {
+	objectMetaFieldsSet := genericregistry.ObjectMetaFieldsSet(&imageNamespace.ObjectMeta, true)
+	specificFieldsSet := fields.Set{
+		"spec.tenantID": imageNamespace.Spec.TenantID,
```
Comment (choujimmy) [inline pkg/business/registry/imagenamespace/strategy.go]: need add "spec.name"

File: pkg/business/registry/rest/rest.go [hunk 7]
```diff
@@ -25,12 +25,14 @@ import (
 	serverstorage "k8s.io/apiserver/pkg/server/storage"
 	restclient "k8s.io/client-go/rest"
 	"tkestack.io/tke/api/business"
-	"tkestack.io/tke/api/business/v1"
+	v1 "tkestack.io/tke/api/business/v1"
```
Comment (choujimmy) [inline pkg/business/registry/rest/rest.go]: remove the `v1` alias

File: pkg/business/util/filter.go [hunk 8]
```diff
@@ -20,9 +20,10 @@ package util
 
 import (
 	"context"
+
 	"k8s.io/apimachinery/pkg/api/errors"
 	"tkestack.io/tke/api/business"
-	"tkestack.io/tke/api/business/v1"
+	v1 "tkestack.io/tke/api/business/v1"
```
Comment (choujimmy) [inline pkg/business/util/filter.go]: remove the `v1` alias

File: pkg/business/registry/rest/rest.go (hunk 7 above)
Comment (defydahui) [reply pkg/business/registry/rest/rest.go]: have used "businessv1" instead of "v1"

File: pkg/business/util/filter.go (hunk 8 above)
Comment (defydahui) [reply pkg/business/util/filter.go]: have used "businessv1" instead of "v1"

**Pull request merged**